src/Workspaces/Core/Portable/Workspace/Host/TemporaryStorage/TrivialTemporaryStorageService.cs

```diff
@@ -14,6 +14,7 @@
 namespace Microsoft.CodeAnalysis
 {
     [ExportWorkspaceService(typeof(ITemporaryStorageService)), Shared]
+    [Export(typeof(TrivialTemporaryStorageService))]
```

Comment:
❔ Why is this getting exported? Exporting this duplicates the export already being done by the factory.

Reply:
I would suggest these changes:

1. Do not export this type
2. Change the constructor to be private (and no attributes)
3. Define a static readonly field `Instance`

Then in the factory, just return `TrivialTemporaryStorageService.Instance` if that's the one you need.